frame/timestamp/src/lib.rs

```diff
@@ -96,54 +96,83 @@ mod benchmarking;
 pub mod weights;
 
 use sp_std::{result, cmp};
-use sp_inherents::{ProvideInherent, InherentData, InherentIdentifier};
+use sp_inherents::InherentData;
 #[cfg(feature = "std")]
 use frame_support::debug;
-use frame_support::{
-	Parameter, decl_storage, decl_module,
-	traits::{Time, UnixTime, Get},
-	weights::{DispatchClass, Weight},
-};
+use frame_support::traits::{Time, UnixTime};
 use sp_runtime::{
 	RuntimeString,
 	traits::{
 		AtLeast32Bit, Zero, SaturatedConversion, Scale,
 	}
 };
-use frame_system::ensure_none;
 use sp_timestamp::{
 	InherentError, INHERENT_IDENTIFIER, InherentType,
 	OnTimestampSet,
 };
 pub use weights::WeightInfo;
 
-/// The module configuration trait
-pub trait Config: frame_system::Config {
-	/// Type used for expressing timestamp.
-	type Moment: Parameter + Default + AtLeast32Bit
-		+ Scale<Self::BlockNumber, Output = Self::Moment> + Copy;
+pub use pallet::*;
 
-	/// Something which can be notified when the timestamp is set. Set this to `()` if not needed.
-	type OnTimestampSet: OnTimestampSet<Self::Moment>;
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use super::*;
 
-	/// The minimum period between blocks. Beware that this is different to the *expected* period
-	/// that the block production apparatus provides. Your chosen consensus system will generally
-	/// work with this to determine a sensible block time. e.g. For Aura, it will be double this
-	/// period on default settings.
-	type MinimumPeriod: Get<Self::Moment>;
+	/// The pallet configuration trait
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Type used for expressing timestamp.
+		type Moment: Parameter + Default + AtLeast32Bit
+			+ Scale<Self::BlockNumber, Output = Self::Moment> + Copy;
 
-	/// Weight information for extrinsics in this pallet.
-	type WeightInfo: WeightInfo;
-}
+		/// Something which can be notified when the timestamp is set. Set this to `()` if not needed.
+		type OnTimestampSet: OnTimestampSet<Self::Moment>;
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
 		/// The minimum period between blocks. Beware that this is different to the *expected* period
 		/// that the block production apparatus provides. Your chosen consensus system will generally
 		/// work with this to determine a sensible block time. e.g. For Aura, it will be double this
 		/// period on default settings.
-		const MinimumPeriod: T::Moment = T::MinimumPeriod::get();
+		#[pallet::constant]
+		type MinimumPeriod: Get<Self::Moment>;
+
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
```

Comment:
The implicit syntax is now slightly recommended over the explicit one
```suggestion
	pub struct Pallet<T>(_);
```